rfcs/000-ml-feature-discovery.md

```diff
@@ -0,0 +1,183 @@
+- Last updated: timestamp for when the feature was most recently updated
+- Usage: number of accesses to the feature, usage in models out of scope for MVP since we are not indexing models yet
+- Upstream resources: upstream lineage in the form of tables, could be expanded to anything else in the future (column, service, pipeline)
+#### Data Pattern Metadata
+Metadata that describes qualities of the data in this feature.
+- Data sample: a sample of the data which the feature contains shown as a table
+- Statistics: any stats relevant to the feature data
+
+
+## UI/UX-level Explanation
+_**Note:** The pink text is only visible in the mocks to call out design details and changes._
+### Home Page
+![Table Lineage Preview](../assets/036/home-page.png)
+##### Expanded Search Bar
+![Table Lineage Preview](../assets/036/home-page-expanded.png)
+### Advanced Search Page
+![Table Lineage Preview](../assets/036/adv-search.png)
+
+### Feature Details Page
+##### Sample Tab
+![Table Lineage Preview](../assets/036/sample-tab.png)
+##### Statistics Tab
+![Table Lineage Preview](../assets/036/stats-tab.png)
+##### Upstream Tab
+![Table Lineage Preview](../assets/036/upstream-tab.png)
+##### Generation Code Tab
+![Table Lineage Preview](../assets/036/generation-tab.png)
+
+## Reference-level Explanation (aka Technical Details)
+
+### Amundsen Databuilder
+The databuilder library will be expanded to include a feature Model that includes the following:
+#### Nodes
+- Feature node:
+    - Key
+    - Name
+    - Version
+    - Feature group
+    - Entity
+    - Data type
+    - Last updated
+#### Relationships
+Here we show the relationships between feature and other nodes, as well as the metadata that this represents: 
+- Feature <FEATURE/FEATURE_OF> Database
+    - Availability
+- Feature <COLUMN/COLUMN_OF> Column <> Badge
+    - Partition column
+    - Note: this column key will have feature key in key instead of a table key
+- Feature <DESCRIPTION/DESCRIPTION_OF> Description
+- Feature <DESCRIPTION/DESCRIPTION_OF> Programmatic_Description
+- Feature <QUERY_OF/HAS_QUERY> Query
+    - Code snippet/SQL: relating it to query from dashboard because we need an external link too
+- Feature <STAT/STAT_OF> Stat
+- Feature <HAS_BADGE/BADGE_FOR> Badge
+    - contains PII
+- Feature <OWNER_TAG/TAGGED_BY_OWNER> Tag
+    - Owner Tags, tag type can now be “owner”, not just default
+- Feature <TAG/TAGGED_BY> Tag
+    - Amundsen User Tags
+- Feature <OWNER/OWNER_OF> User
+- Feature <WATERMARK> Watermark
+    - Date range
+
+### Amundsen Common
+Amundsen common will have the folllowing schema which represents the feature response between servies:
+```
+class Feature:
+   key: str  # ex: test_feature_group_name/test_feature_name/1.2.0
+   name: str
+   version: str  # ex: "1.2.0"
+   status: str
+   feature_group: str
+   entity: Optional[List[str]]
+   data_type: Optional[str]
+   availability: List[str]
```

Comment:
I think this is already included in upstream resources. The issue with making it the same in availability is that we might lost important metadata if the resource is not listed in Amundsen, for example we don't index dynamo stuff yet but we want to tell ML engs that this feature is in fact available there.

Reply:
I see your point, there could be a lot of information loss here. 

Given that this field just a string and the upstream information is available as you mentioned, I'm wondering if this needs to be a first class object or this is an area where the implementer could use prog desc to provide this info. I'd be curious how much others in the community see this being used.

Reply:
Since it is a list just having it be [] and not provide the information wouldn't really make a big difference, we could make it a prog description too. This is the use case we had internally for it, if it doesn't come up for anyone else then maybe we can remove it: https://github.com/amundsen-io/rfcs/pull/36#discussion_r627658672 what do you think?

Reply:
Sounds good, lets see if anyone else in the community can chime in to share an external perspective.